wagtail/admin/views/generic/chooser.py

```diff
@@ -174,6 +175,15 @@ def get_filter_form(self):
         return FilterForm(self.request.GET)
 
     def filter_object_list(self, objects):
+        filters = {}
+        for filter in self.url_filter_parameters:
+            try:
+                filters[filter] = self.request.GET[filter]
+            except KeyError:
+                pass
+        if filters:
+            objects = objects.filter(**filters)
```

Comment:
I wonder we should have a similar mechanism to the filter form's `is_searching` (or maybe incorporate this parameter filter logic into the filter form instead?) when `filters` here is not empty.

Without it, the template will render as if there are no instances of the model, which could be misleading:

<img width="821" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/58d9321d-43c1-4303-ab9a-d4cf6589355e">

It seems this was already the case (not sure, haven't checked) when form filters are used though, so I wouldn't mind if this is addressed separately.

Reply:
I agree that this message isn't great, but I'm not sure that adding a distinct message for the case where filters are in force is the way to fix it. From the user's perspective the outcome is the same: there's nothing available for you to choose here.

I'd propose just changing this message to "There are no results". Referring to the act of creating items just seems out of place and irrelevant to the user's current workflow - especially as in the general case, the person receiving the message probably isn't responsible for creating them. It's like going to a shop and asking "do you have any bananas?" and having them reply "No bananas have been delivered to us." :-)

(I suspect this message is actually a hangover from other listings where we provide a "Would you like to create one now?" link...)

The message is technically correct for both filtered and unfiltered listings - nobody has created any items that fulfil the criteria for appearing in this list - it's just that with filtered listings, the weirdness of phrasing it in that way just becomes more apparent.

Reply:
I agree, I think we can benefit from the simplicity of just changing the message 👍